daemon/mgr/container.go

```diff
@@ -310,9 +310,8 @@ func (mgr *ContainerManager) Create(ctx context.Context, name string, config *ty
 
 	// set default log driver and validate for logger driver
 	if config.HostConfig.LogConfig == nil {
-		config.HostConfig.LogConfig = &types.LogConfig{
-			LogDriver: types.LogConfigLogDriverJSONFile,
-		}
+		defaultConfig := mgr.Config.DefaultLogConfig
+		config.HostConfig.LogConfig = &defaultConfig
```

Comment:
The request maybe come from swarm. If it does, the `config.HostConfig.LogConfig` is not nil but the value is empty. For this case, the `config.HostConfig.LogConfig.LogDriver` is empty and `config.HostConfig.LogConfig. LogOpts` is nil. 

It isn't good practice in API design. However, we need to compatible with legacy system. We need to add other check that  `config.HostConfig.LogConfig` is empty or not. WDTY?